src/opcodes/tests/test_opcodes.cairo

```diff
@@ -1068,3 +1068,22 @@ fn test_op_1negate() {
     let expected_stack = array![ScriptNum::wrap(-1)];
     assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
 }
+
+#[test]
+fn test_op_ifdup() {
```

Comment:
Could you include 2 tests:
1. Top stack element is 0 so no dup is done
2. Top stack element is non-0 so dup is done ( like this one )

Reply:
Alright